pkg/smartcontract/binding/generate.go

```diff
@@ -234,16 +234,11 @@ func TemplateFromManifest(cfg Config, scTypeConverter func(string, smartcontract
 
 		// Consider `perform(a)` and `perform(a, b)` methods.
 		// First, try to export the second method with `Perform2` name.
-		// If `perform2` is already in the manifest, use `perform_2` with as many underscores
-		// as needed to eliminate name conflicts. It will produce long names in certain circumstances,
-		// but if the manifest contains lots of similar names with trailing underscores, delicate naming
-		// was probably not the goal.
+		// If `perform2` is already in the manifest, use `perform3` with uprising suffix as many times
+		// as needed to eliminate name conflicts. If `perform3` is already in the manifest, use `perform4` etc.
 		name := m.Name
-		if v, ok := seen[name]; !ok || v {
-			suffix := strconv.Itoa(len(m.Parameters))
```

Comment:
Now the only problem is that this number has meant something previously (`Sum3` is "Sum with three parameters"). So we can either make it meaningless (sufficient to differentiate functions, like now), or we can try `m.Name + strconv.Itoa(len(m.Parameters))` and then add `X` to it until it works. Up to you, @AnnaShaleva.

Reply:
Maybe try the letter? It's convenient to distinguish functions by arguments number.

Reply:
Or maybe we don't even need this and it's OK to have the methods with different names. Let's keep the current version.